torch/distributed/pipelining/schedules.py

```diff
@@ -19,6 +19,7 @@
     "PipelineScheduleSingle",
     "PipelineScheduleMulti",
     "Schedule1F1B",
+    "ScheduleFlexibleInterleaved1F1B",
```

Comment:
we should add a unit test case that covers this schedule.  The best one to follow is the 'grad' test.  it confirms the schedule correctly runs forward and backward and populates grads onto the model matching a reference model.  i think its in test/pipelining/test_schedule.py